src/hypothesis/strategies.py

```diff
@@ -982,32 +987,61 @@ def fractions(min_value=None, max_value=None, max_denominator=None):
 
     If max_denominator is not None then the absolute value of the denominator
     of any generated values is no greater than max_denominator. Note that
-    max_denominator must be at least 1.
+    max_denominator must be None or a positive integer, and the denominator
```

Comment:
Yes. I gave him a _full_ description of the problem :) But I'm not optimistic that there is any insight that can be offered beyond the obvious that iff
* the max denominator is less than the denominator of each endpoint, AND
* the range between the endpoints is less than 1/the max denominator

then it _may_ be computationally inexpensive to decide if there is an example.